docs/security.md

```diff
@@ -0,0 +1,154 @@
+# Security
+
+## Overview
+
+### Role based security
+
+Role-based security allows Virtocommerce administrators to group users into roles for security purposes. Each role can be defined with specific granular permissions to allow access to different functionality within the software.
+
+Security system is composed of three main entities
+
+1. Permissions is a basic unit in security system. It describes particular right of action and has a string presentation which is used in permission checks.
+
+1. Roles are used to collect permissions that define a particular function within the platform, according to a particular scope. Roles can grant permissions to various functions within platform. A role is basically just a collection of permissions. Users that are assigned to the role will inherit these permissions.
+
+1. Scopes are used to restrict permissions to concrete conditions or objects.
+
+### Bounded scope
+
+Used in situation when you need to restrict user access to particular objects (manually selected or selected by condition). For example: Restrict access to the specified stores or allow update orders only by responsible users.
+
+### Users
+
+Users are individuals working with **Virto Commerce** management application and services.
+
+In order to manage **Virto Commerce** functionality each user must be assigned to at least one role. Each role provides the user with certain access permissions.These permissions allow or restrict the user's access to functionalities within the **Virto Commerce** client application.
+
+Common Roles are provided by default with each Virto Commerce installation. User roles can be created or edited to provide their staff with permissions reflecting only the tasks they perform within the **Virto Commerce** management application.
+
+After installation, there is one user in the system with Super User (administrator) privileges. Only administrator users have permission to manage users and roles.
+
+### Roles and Permissions
+
+Permissions are controlled by assigning Roles to users. A Role is a collection of permissions. A Role can be assigned to multiple users. Each user can have more than one assigned Role.
+
+Through the combination of assigned Roles, you can ensure that users only have access to the information and functionality they need.
+This model provides considerable flexibility, but in complex environments with multiple stores and catalogs, each with their own distinct sets of users, it is critical to establish best practices for managing permissions.
```

Comment:
What is "best practices for managing permissions"?

Reply:
deleted!